cdap-api/src/main/java/co/cask/cdap/internal/schedule/StreamSizeSchedule.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.internal.schedule;
+
+import co.cask.cdap.api.schedule.Schedule;
+
+/**
+ * Defines a schedule based on data availability in a stream.
+ */
+public final class StreamSizeSchedule extends Schedule {
+
+  private final String streamName;
+  private final int dataTriggerMB;
+
+  public StreamSizeSchedule(String name, String description, String streamName, int dataTriggerMB) {
+    super(name, description);
+    this.streamName = streamName;
+    this.dataTriggerMB = dataTriggerMB;
+  }
+
+  /**
+   * @return Name of the stream this {@link StreamSizeSchedule} is based on
+   */
+  public String getStreamName() {
```

Comment:
Namespace will be inferred at runtime using the program that the schedule is running on. Users don't specify the namespace when configuring an app.